dummyfs/dummyfs.c

```diff
@@ -876,8 +958,36 @@ int dummyfs_mount(void **ctx, const char *data, unsigned long mode, oid_t *root)
 	o->oid.port = fs->port;
 	o->mode = S_IFDIR | 0666;
 
-	memcpy(&rootdir, &o->oid, sizeof(oid_t));
+	rootdir = o->oid;
+	fs->parent = rootdir;
+
+	if (data != NULL) {
+		abspath = resolve_path(data, NULL, 1, 0);
+		if (abspath == NULL) {
+			return -ENOMEM;
+		}
+		if (lookup(abspath, &fs->parent, NULL) < EOK) {
+			free(abspath);
+			dev_cleanup(fs);
+			object_cleanup(fs);
+			resourceDestroy(fs->mutex);
+			free(fs);
+			return -ENOENT;
+		}
+
+		fs->mountpt = realloc(abspath, strlen(abspath) + 1);
+		if (fs->mountpt == NULL) {
+			dev_cleanup(fs);
```

Comment:
I believe you should free(abspath) here too. 
If realloc fails it should left original memory untouched.